### DISCUSSION TITLE
Retention Policy using OR instead of AND

### DISCUSSION THREAD

**Discussion by giblets** [datalust/seq-tickets#1269] [Help]
Have implemented Seq where I'm working currently.  It's great, everybody loves it!

One problem though is that it appears that Retention policy is using an OR instead of AND for purging data.
![image](https://user-images.githubusercontent.com/590086/127696291-1d4f6a73-4088-43a9-af8c-ad83d28c88d7.png)

The above policy was deleting every "Information" level log message regardless of the the first signal in the filter.  ...We were able to get around the problem by making a custom signal specifically that included "Information" messages as part of it.  But as far as we can tell the functionality is broken and doesn't adhere to the "Events that match all of the selected signals will be deleted by the policy".  ...For us it was deleting every "Information" message after 1 hour.  

If there are any logs we can check to provide more information please let us know.

**Comment by nblumhardt** [comment]
Hi Glenn,

Thanks for the heads-up, and sorry about the poor UX here, we need to make that help text clearer!

Normally, signals are "or"ed together when they're in the same signal group. The two selected signals are sticking together into one block there because they're in the same group, but I'm not sure why that would be. You can see the grouping details for each of the signals back in the signal editor.

If it appears the signals are ungrouped, then you may be hitting a bug that went out in an earlier 2021 release; updating to 2021.2.5647 and refreshing the browser will fix it, if that's the case.

(https://docs.datalust.co/docs/signals#how-signals-are-combined has some info on how signal groups work when combined, but could use some illustration.)

**Comment by giblets** [comment]
Thanks Nicholas!

Our dashboard is looking a lot better since we implemented the workaround.

...We have heavy Data Replication over here at Micrologic and we get
millions of events for it every day.  Usually we throttle the API Key to
only take in Error events, but with the retention policies we've been able
to keep information events for these for an hour.

[image: image.png]

This gives us a really neat look at client per client how replication is
working for them.  (We're in the carwash software business, so all the
client names usually have "Foam", "Splash" and similar words in them :) )

[image: image.png]

Seq and Serilog have been great for creating dashboards that support and
the dev team can monitor to quickly find problems and remediate them.
...At my last place we initially tried Elasticsearch and the UI wasn't
anywhere near as good.  My last place NJPLIGA bought a Seq license too.
...Hoping very much it continues to grow and have success as a product.

Thanks again, and thanks for the great documentation.  We absolutely love
it!!!

Cheers,
-Glenn

**Reply by nblumhardt** [reply]
That's awesome; thanks for the feedback, @giblets! :-)
